### DISCUSSION TITLE
Selecting subset of full document on Load to reduce size?

### DISCUSSION THREAD

**Discussion by kamranayub** [ravendb/ravendb#18210] [Q&A] [answered]
In a query, I can use projection to limit the size of the results payload (and deserialization cost).

I'm wondering if there's anything like that for Load. I didn't see anything in the docs. I didn't test whether providing a different slimmed down class for the type might work.

Basically the context is that I have a known list of IDs and I need to cross reference a property on those documents. Ideally, I wouldn't need to load the full document because this is a bulk operation with possibly thousands of IDs. I was worried about memory consumption and ways to optimize just getting the field(s) I need from the documents returned by the server.

I could use an index but since I have the IDs, I didn't want to worry about staleness.

I thought about streaming, but I think it will still load the full document, right?

**Comment by DejanMilicic** [comment, marked as answer]
You can go with this

```
session.Query<Employee>()
    .Where(x => x.Id.In("employees/1-A", "employees/2-A"))
    .Select(x =>
        new
        {
            First = x.FirstName,
            Last = x.LastName
        })
    .ToList();
```

Filtering on IDs will work directly with key/value store, and will not hit any of the indexes.

**Reply by kamranayub** [reply]
Thanks, I wondered if that was the case. I think the docs mention this somewhere, but it could be good to call this out with a "Load Projections" section or something, since right now projections are only talked about under indexing.
